pymc3/step_methods/hmc/hmc.py

```diff
@@ -85,11 +86,13 @@ def __init__(self, vars=None, path_length=2.,
             The model
         **kwargs : passed to BaseHMC
         """
+        kwargs.setdefault('step_rand', unif)
+        kwargs.setdefault('target_accept', 0.65)
         super().__init__(vars, **kwargs)
         self.path_length = path_length
 
     def _hamiltonian_step(self, start, p0, step_size):
-        path_length = np.random.rand() * self.path_length
```

Comment:
Do we really want to remove this? If so `n_steps = max(1, int(path_length / step_size))` is constant and we dont need to recomputed it in each step any more.

Reply:
https://github.com/pymc-devs/pymc3/blob/master/pymc3/step_methods/hmc/base_hmc.py#L151 takes care of that (using `step_rand`)

Reply:
Oh OK, good catch of the bug.